docs/DEVELOPMENT_GUIDE.md

```diff
@@ -0,0 +1,111 @@
+# Development Guide
```

Comment:
please rename the file to 'development_guide.md' (it's more natural to have these files all lower case).

Reply:
Done.